### DISCUSSION TITLE
Welcome to solis-ha-modbus-cloud Discussions!

### DISCUSSION THREAD

**Discussion by alienatedsec** [alienatedsec/solis-ha-modbus-cloud#4] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, you can comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by magichew** [comment]
Hi

Thanks for setting discussions up. 

I’ve got a solis inverter and data logger as well as HA pulling SolisCloud information, Solar Forecasting and rates from Octopus. 

I’m lurking in the Git hoping for an idiots guide to setting this up to appear from the author or users

I love the idea of more frequent (than 5 minutes) updates into the HA dashboard which if I can implement the work done in this Git would at the very most basic level of integration allow me to automate turning on our 3 pin plug car charger when we’re exporting and turn off automatically. 

Further to that, of course building in smart charge and discharge of the battery based on agile rates. 

I’ve seen in the Facebook group someone appears to have a GUI to control the inverter using this setup which is ideal. I’ve also seen YouTube videos where there are scripts used to control the inverter. The GUI would be more up my stream as I don’t have a programming background. 

I am yet to order the equipment required until I feel confident that I can pull this off myself. 

Thank you for the work done on this so far.

**Reply by alienatedsec** [reply]
Welcome @magichew 

These are the same reasons why I started this project. I knew very little about it, but spending more time on education and its design allowed me to share my work with the community, and make some of it easier to digest (I hope).

> Further to that, of course building in smart charge and discharge of the battery based on agile rates.

I've made something - not using it at the moment as still on a cheap overnight tariff, but will have to jump on it at some point - there will be a section dedicated to scripts and ideas - maybe even a wiki page. Not the best view but below is my initial work and dedicated timeslots for the best pricing. I plan to use it for many other devices. E.g. an immersion heater in winter etc.

![image](https://user-images.githubusercontent.com/73167064/226841071-a84a19c5-ffc1-4fd7-9730-e1135a47064b.png)

> I love the idea of more frequent (than 5 minutes) updates into the HA dashboard which if I can implement the work done in this Git would at the very most basic level of integration allow me to automate turning on our 3 pin plug car charger when we’re exporting and turn off automatically.

I have automation which relies on 15 seconds intervals; therefore, having an update every 5 seconds makes it a perfect smart solution.

Depending on the car charger, that should be already its primary function to charge whenever excess from the PV is to be exported. What is your model?

> I’ve seen in the Facebook group someone appears to have a GUI to control the inverter using this setup which is ideal. I’ve also seen YouTube videos where there are scripts used to control the inverter. The GUI would be more up my stream as I don’t have a programming background.

The ability to control the inverter via HA depends on the way you connect your inverter - I believe this project allowing any Solis Datalogger and the Waveshare Modbus TCP with SolaX integration is the best combination so far. Relying on a datalogger to serve both purposes comes with conditions and limitations.

> I am yet to order the equipment required until I feel confident that I can pull this off myself.

Let's start with that - if you can read the wiring diagram, can solder, and are not afraid of DIY - you are more than half way through

**Reply by magichew** [reply]
Thank you for taking the time to write such a thorough response. I am going to give this ago, it's not beyond my skill set, you've given me the confidence to do it.

**Comment by magichew** [comment]
I’m very interested in your set up. The installation looks very neat in the box with the rail mount. Can you also add an image to the read me of how you have finished the connector at the inverter end?

**Reply by alienatedsec** [reply]
@magichew I don't think that is necessary as there will be nothing unusual there.

The cable goes into the plug, which is the same as your datalogger.

**Reply by alienatedsec** [reply]
@magichew I guess there is nothing interesting here

![image](https://user-images.githubusercontent.com/73167064/228971818-8158077f-6612-4495-a4d6-ec62ca0feeca.png)

**Comment by magichew** [comment]
No that is definitely interesting. I had seen images in the Solis Inverter Facebook group of bare wires soldered to the pins of the plug and didn't like the look of it. Obviously that is a sleeve that is part of the plug? Have you used CAT5 cable? Did you use crimp connectors to slide on to / in to the socket or solder directly?

**Reply by alienatedsec** [reply]
I used an ethernet cable (CAT6 but any would do) as it comes with a twisted pair cabling by design and works very well with RS485 networks. You will need to solder as there is no option to use a crimp tool unless other plugs are used.

E.g., https://www.thingiverse.com/thing:5338738

The plug comes with the plastic cover, but the sleeve - a black electrical tape on the cable - is to keep the cable in place and protect it when the plastic cover and the cabling holder are screwed/tightened together with the plug.
